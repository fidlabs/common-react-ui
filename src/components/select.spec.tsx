import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import {
  Select,
  SelectContent,
  SelectGroup,
  SelectItem,
  SelectLabel,
  SelectTrigger,
  SelectValue,
} from "./select";

describe("Select component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Select>
        <SelectTrigger>
          <SelectValue placeholder="Select option" />
        </SelectTrigger>
        <SelectContent>
          <SelectGroup>
            <SelectLabel>Options</SelectLabel>
            <SelectItem value="option-1">Option 1</SelectItem>
            <SelectItem value="option-2">Option 2</SelectItem>
          </SelectGroup>
        </SelectContent>
      </Select>
    );

    expect(container).toMatchSnapshot();
  });
});
