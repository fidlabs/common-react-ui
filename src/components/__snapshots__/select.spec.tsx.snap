// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Select component > renders correctly 1`] = `
<div>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:r0:"
    aria-expanded="false"
    class="flex h-10 w-full items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background data-[placeholder]:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-dodger-blue focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      Select option
    </span>
    <svg
      aria-hidden="true"
      class="lucide lucide-chevron-down h-4 w-4 opacity-50"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m6 9 6 6 6-6"
      />
    </svg>
  </button>
</div>
`;
